File: src/main/java/org/artembogomolova/demo/webapp/dao/repo/auth/IRoleRepository.java
package org.artembogomolova.demo.webapp.dao.repo.auth;

import org.artembogomolova.demo.webapp.domain.auth.Role;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface IRoleRepository extends CrudRepository<Role, Long> {

  Role findByName(String name);
}

File: app/src/main/kotlin/org/artembogomolova/demo/webapp/main/dao/repo/IRoleRepository.kt
package org.artembogomolova.demo.webapp.main.dao.repo

import org.artembogomolova.demo.webapp.main.domain.auth.Role
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
interface IRoleRepository : CrudRepository<Role?, Long?> {
    fun findByName(name: String?): Role?
}